fuel-tests/tests/snapshot.rs

```diff
@@ -0,0 +1,74 @@
+use fuel_core::database::Database;
+use fuel_core::{
+    config::{
+        chain_config::{CoinConfig, ContractConfig, StateConfig},
+        Config,
+    },
+    service::FuelService,
+};
+use fuel_core_interfaces::common::{
+    fuel_types::{Address, Bytes32, Salt},
+    fuel_vm::prelude::{AssetId, Contract, ContractId, InterpreterStorage, Storage},
+};
+
+#[tokio::test]
+async fn snapshot_state_config() {
+    let mut db = Database::default();
+
+    let owner = Address::default();
+    let asset_id = AssetId::new([1u8; 32]);
+
+    // Extract later for a test case
+    let contract = Contract::default();
+    let id = ContractId::new([12; 32]);
+
+    // setup config
+    let mut config = Config::local_node();
+    config.chain_conf.initial_state = Some(StateConfig {
+        height: None,
+        contracts: Some(vec![ContractConfig {
+            code: vec![8; 32],
+            salt: Salt::new([9; 32]),
+            state: None,
+            balances: None,
```

Comment:
test should cover state and balances as well, as it would reveal current problems in the implementation